va/va.go

```diff
@@ -534,6 +556,80 @@ func (va VAImpl) fetchHTTP(identifier string, token string) ([]byte, string, *ac
 	return body, url.String(), nil
 }
 
+func (va VAImpl) getTXTEntry(name string) ([]string, error) {
+	ctx, cancelfunc := context.WithTimeout(context.Background(), validationTimeout)
+	defer cancelfunc()
+
+	if va.customResolverAddr == "" {
+		return net.DefaultResolver.LookupTXT(ctx, name)
+	}
+
+	txts := []string{}
+	client := new(dns.Client)
+	message := new(dns.Msg)
+	message.SetQuestion(dns.Fqdn(name), dns.TypeTXT)
+	in, _, err := client.ExchangeContext(ctx, message, va.customResolverAddr)
+
+	if err != nil || in.Rcode != dns.RcodeSuccess {
+		return txts, err
+	}
+
+	for _, record := range in.Answer {
+		if t, ok := record.(*dns.TXT); ok {
+			txts = append(txts, t.Txt...)
+		}
+	}
+
+	return txts, nil
+}
+
+func (va VAImpl) resolveIP(name string) ([]string, error) {
+	ctx, cancelfunc := context.WithTimeout(context.Background(), validationTimeout)
+	defer cancelfunc()
+
+	if va.customResolverAddr == "" {
+		return net.DefaultResolver.LookupHost(ctx, name)
+	}
+
+	addrs := []string{}
+	parsed := net.ParseIP(name)
```

Comment:
Is this to handle the case where `resolveIP` is called with `name` as an IP address encoded as a string? I'm not sure I understand this parsing check.

Reply:
It is for this case indeed. I add an inline comment.

Reply:
Let's put this check at the top of the function then, since it short-circuits the rest.

Also, it would be clearer to make the return statement be `return []net.IP{parsed}`.